kernel/basic.ml

```diff
@@ -41,7 +41,13 @@ let mk_ident     = WS.merge shash
 
 let mk_mident md =
   let base = Filename.basename md in
-  try Filename.chop_extension base with _ -> base
+  let le = String.length base in
+  if le > 3 &&
+     String.get base (le-3) = '.' &&
+     String.get base (le-2) = 'd' &&
+     String.get base (le-1) = 'k'
```

Comment:
Is it really better to do 3 accesses to one character, rather than just `String.sub base (le-3) 3 = ".dk"`

Reply:
Actually, you can do better:
```ocaml
Filename.(if check_suffix base ".dk" then chop_suffix base ".dk" else base)
```